### PR TITLE
Switch to Reference Design Template codebase

### PR DESCRIPTION
**Pull request by MarkoPura**
Golioth OpenThread Demo based on the Reference Design Template with support for nRF52840 DK and Adafruit nRF52840 Feather commercially available development boards.

Need to create a release and update the link in README to projects.golioth.io.

### nRF52840DK
  * [x] Shell/Credentials
  * [x] Connection
  * [x] Logs
  * [x] State
  * [x] Stream
  * [x] Button press
  * [x] Settings (Loop Delay)
  * [x] RPC (log level)
  * [x] RPC(reboot)
  * [x] OTA Update

### Adafruit Feather 52840
  * [x] Shell/Credentials
  * [x] Connection
  * [x] Logs
  * [x] State
  * [x] Stream
  * [x] Button press
  * [x] Settings (Loop Delay)
  * [x] RPC (log level)
  * [x] RPC(reboot)
  * [x] OTA Update

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  pull_request:
+
+  push:
+
+jobs:
+  test_build:
+    uses: ./.github/workflows/build_zephyr.yml
+    with:
+      ZEPHYR_SDK: 0.16.3
+      BOARD: aludel_mini_v1_sparkfun9160_ns
```
Comment (szczys) [inline .github/workflows/test.yml]: Should this target the 52840 for automated build tests?

```suggestion
      BOARD: nrf52840dk_nrf52840
```

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                required: true
+                default: 'v0.0.0'
+                type: string
+
+jobs:
+    build-binaries:
+        strategy:
+            matrix:
+              ZEPHYR_SDK: [0.16.3]
+              BOARD: ["nrf9160dk_nrf9160_ns","aludel_mini_v1_sparkfun9160_ns"]
```
Comment (szczys) [inline .github/workflows/release.yml]: Should this target the two boards you have listed as node hardware?

File: .github/workflows/build_zephyr.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  workflow_dispatch:
+    inputs:
+      ZEPHYR_SDK:
+        required: true
+        type: string
+        default: 0.16.3
+      BOARD:
+        required: true
+        type: string
+        default: aludel_mini_v1_sparkfun9160_ns
```
Comment (szczys) [inline .github/workflows/build_zephyr.yml]: Maybe this should default to the 52840dk?

File: README.rst [hunk 4]
```diff
@@ -62,45 +72,209 @@ by changing these lines in the ``prj.conf`` configuration file, e.g.:
...
+   ``<your_zephyr_board_id>`` with the actual Zephyr board from the table above
+   that matches your hardware.
+
+   In addition, replace ``<your.semantic.version>`` with a `SemVer`_-compliant
+   version string (e.g. ``1.2.3``) that will be used by the DFU service when
+   checking for firmware updates.
+
+.. code-block:: text
+
+   $ (.venv) west build -p -b <your_zephyr_board_id> app --
```
Comment (szczys) [inline README.rst]:
```suggestion
   $ (.venv) west build -p -b <your_zephyr_board_id> app
```

File: README.rst [hunk 5]
```diff
@@ -62,45 +72,209 @@ by changing these lines in the ``prj.conf`` configuration file, e.g.:
...
+
+For example, to build firmware for the `Nordic nRF52840 DK`_-based follow-along hardware:
+
+.. code-block:: text
+
+   $ (.venv) west build -p -b nrf52840dk_nrf52840 app --
```
Comment (szczys) [inline README.rst]:
```suggestion
   $ (.venv) west build -p -b nrf52840dk_nrf52840 app
```

File: README.rst [hunk 6]
```diff
@@ -1,58 +1,68 @@
...
-********
+Golioth OpenThread Demo
+#######################
 
-A demonstration of a Zephyr device connecting to Golioth over IPv6 via Thread Protocol.
+This repository contains the firmware source code and `pre-built release
+firmware images <releases_>`_ for the Golioth OpenThread Demo.
 
-This is a standalone repository that will download all required files and dependencies. This directory will take up a good amount of room, as it will contain the latest Nordic NCS in the ``deps`` repository..
+[To be updated] The full project details are available on the `Golioth Thread Demo Project Page`_.
```
Comment (szczys) [inline README.rst]:
```suggestion
The full project details are available on the `Golioth Thread Demo Project Page`_.
```

Maybe we should update this now as we know what the projects.golioth.io URL will be

File: README.rst [hunk 7]
```diff
@@ -62,45 +72,209 @@ by changing these lines in the ``prj.conf`` configuration file, e.g.:
...
+   git merge template_v1.0.0
+
+   # Resolve merge conflicts if necessary
+   git add resolved_files
+   git commit
+
+.. _Golioth Console: https://console.golioth.io
+.. _GL-S200 Thread Border Router: https://www.gl-inet.com/products/gl-s200/
+.. _Nordic nRF52840 DK: https://www.nordicsemi.com/Products/Development-hardware/nRF52840-DK
+.. _Golioth Thread Demo Project Page: https://golioth.github.io/golioth-openthread-demo-docs
```
Comment (szczys) [inline README.rst]:
```suggestion
.. _Golioth Thread Demo Project Page: https://projects.github.io//reference-designs/openthread-demo/
```

File: boards/adafruit_feather_nrf52840.conf [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+CONFIG_SETTINGS_RUNTIME=y
+CONFIG_GOLIOTH_SAMPLE_PSK_SETTINGS=y
+CONFIG_GOLIOTH_SAMPLE_SETTINGS_AUTOLOAD=y
+CONFIG_GOLIOTH_SAMPLE_SETTINGS_SHELL=y
+
+CONFIG_FLASH=y
+CONFIG_FLASH_MAP=y
+CONFIG_NVS=y
+
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=8192
```
Comment (szczys) [inline boards/adafruit_feather_nrf52840.conf]: This seems like a really large stack. Do we need that much memory?

File: boards/nrf52840dk_nrf52840.conf [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+CONFIG_SETTINGS_RUNTIME=y
+CONFIG_GOLIOTH_SAMPLE_PSK_SETTINGS=y
+CONFIG_GOLIOTH_SAMPLE_SETTINGS_AUTOLOAD=y
+CONFIG_GOLIOTH_SAMPLE_SETTINGS_SHELL=y
+
+CONFIG_FLASH=y
+CONFIG_FLASH_MAP=y
+CONFIG_NVS=y
+
+CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=8192
```
Comment (szczys) [inline boards/nrf52840dk_nrf52840.conf]: Same question as above... do we need this much memory?

File: src/main.c [hunk 10]
```diff
@@ -1,205 +1,118 @@
...
+#include <zephyr/kernel.h>
 
-#include <net/golioth/system_client.h>
+#include <zephyr/drivers/gpio.h>
 
-#include <net/openthread.h>
+#include <zephyr/net/openthread.h>
 #include <openthread/thread.h>
+#include <zephyr/net/coap.h>
+#include <zephyr/net/socket.h>
```
Comment (szczys) [inline src/main.c]: nit: This might be tidier if the includes were alphabetized.

**Review by szczys — APPROVED** [review]
Some changes to the workflows and suggestions in the README, but all looks good in general. This is a huge chunk of work, great job!

File: boards/adafruit_feather_nrf52840.conf (hunk 8 above)
Comment (MarkoPura) [reply boards/adafruit_feather_nrf52840.conf]: We don't, remainders from development. Changed to `2048`

**Pull request merged**